### PR TITLE
Wip 11442

### PR DESCRIPTION
**Pull request by yehudasa**
A fix and a tool to recover from issue #11442. The fix will make it so that objects will revert to using the old locator behavior when object names start in underscore.
The tool (via radosgw-admin) will convert objects that were created since hammer that start with underscore.

### CODE REVIEW

File: src/rgw/rgw_admin.cc [hunk 1]
```diff
@@ -1128,6 +1247,10 @@ int main(int argc, char **argv)
       // do nothing
     } else if (ceph_argparse_binary_flag(args, i, &fix, NULL, "--fix", (char*)NULL)) {
       // do nothing
+    } else if (ceph_argparse_binary_flag(args, i, &remove_bad, NULL, "--remove_bad", (char*)NULL)) {
```
Comment (jdurgin) [inline src/rgw/rgw_admin.cc]: --remove-bad would be more consistent

File: src/rgw/rgw_admin.cc [hunk 2]
```diff
@@ -1128,6 +1247,10 @@ int main(int argc, char **argv)
...
+      // do nothing
+    } else if (ceph_argparse_binary_flag(args, i, &check_head_obj_locator, NULL, "--check-head-obj-locator", (char*)NULL)) {
```
Comment (jdurgin) [inline src/rgw/rgw_admin.cc]: should add these to usage and man page

Comment (yehudasa) [inline src/rgw/rgw_admin.cc]: @jdurgin really not. Should add this to the specific bug it fixes.

**Comment by yehudasa** [comment]
@jdurgin pushed an update for the --remove_bad inconsistency. I don't think we wan to add this to the man page / usage.

**Pull request merged**